packages/react-core/src/demos/examples/AlertGroup/AlertGroupToastWithNotificationDrawer.tsx

```diff
@@ -0,0 +1,365 @@
+      Mark all read
+    </DropdownItem>,
+    <DropdownItem key="clearAll" onClick={removeAllNotifications} component="button">
+      Clear all
+    </DropdownItem>
+  ];
+
+  const notificationDrawerDropdownItems = (key: React.Key) => [
+    <DropdownItem key="markRead" component="button" onClick={() => markNotificationRead(key)}>
+      Mark as read
+    </DropdownItem>,
+    <DropdownItem key="action" component="button" onClick={() => removeNotification(key)}>
+      Clear
+    </DropdownItem>
+  ];
+
+  const notificationDrawer = (
+    <NotificationDrawer>
+      <NotificationDrawerHeader count={getUnreadNotificationsNumber()} onClose={() => setDrawerExpanded(false)}>
+        <Dropdown
+          onSelect={onDropdownSelect}
+          toggle={
+            <KebabToggle
+              onToggle={isActive => onDropdownToggle('dropdown-toggle-id-0', isActive)}
+              id="dropdown-toggle-id-0"
+            />
+          }
+          isOpen={openDropdownKey === 'dropdown-toggle-id-0'}
+          isPlain
+          dropdownItems={notificationDrawerActions}
+          id="notification-drawer-0"
+          position={DropdownPosition.right}
+        />
+      </NotificationDrawerHeader>
+      <NotificationDrawerBody>
+        {notifications.length !== 0 && (
+          <NotificationDrawerList>
+            {notifications.map(({ key, variant, title, srTitle, description, timestamp }) => (
+              <NotificationDrawerListItem key={key} variant={variant} isRead={isNotificationRead(key)}>
+                <NotificationDrawerListItemHeader variant={variant} title={title} srTitle={srTitle}>
+                  <Dropdown
+                    position={DropdownPosition.right}
+                    onSelect={onDropdownSelect}
+                    toggle={<KebabToggle onToggle={isActive => onDropdownToggle(key, isActive)} id={key.toString()} />}
+                    isOpen={openDropdownKey === key}
+                    isPlain
+                    dropdownItems={notificationDrawerDropdownItems(key)}
+                    id={key.toString()}
+                  />
+                </NotificationDrawerListItemHeader>
+                <NotificationDrawerListItemBody timestamp={timestamp}> {description} </NotificationDrawerListItemBody>
+              </NotificationDrawerListItem>
+            ))}
+          </NotificationDrawerList>
+        )}
+        {notifications.length === 0 && (
+          <EmptyState variant={EmptyStateVariant.full}>
+            <EmptyStateIcon icon={SearchIcon} />
+            <Title headingLevel="h2" size="lg">
+              No notifications found
+            </Title>
+            <EmptyStateBody>There are currently no notifications.</EmptyStateBody>
+          </EmptyState>
+        )}
+      </NotificationDrawerBody>
+    </NotificationDrawer>
+  );
+
+  return (
+    <DashboardWrapper
+      header={<DashboardHeader notificationBadge={notificationBadge} />}
```

Comment:
@mcoker That definitely is possible and I agree that it would be more consistent.. However, I cannot think about any neat solution. We could access the `NotificationDrawerListItem`s through children of `notificationDrawer` and extract the required data from there, but that approach requires to go quite deep into the tree and there won't be any guarantee that the required props (`variant`, `isRead`) are set. I'd suggest to leave this as it is for now and maybe open this in a new issue, @thatblindgeye what do you think?

Reply:
I haven't yet tried looking into it more closely, but I would agree that this would make sense as a follow-up issue.